[RM Ch04] Underlay and Overlay Networking concepts (#1772)
cntt-n/CNTT: doc/ref_model/chapters/chapter04.md

 - Software Defined Underlay and Overlay Networking concepts 
 - Programmable Networking Fabric concept

<a name="4.3.1.1"></a>
#### 4.3.1.1 Underlay and Overlay Networking concepts

> The ETSI NFV model divide networking in an Underlay and an Overlay Network layer. The purpose with this layering is to ensure separation of the Virtualisation Tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network in between all Ethernet connected hardware (HW) devices.

Reviewer: ```suggestion
The ETSI NFV model divides networking into an Underlay and an Overlay Network layer. The purpose of this layering is to ensure separation of the Virtualisation Tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network (connected hardware (HW) devices).
```

> The Overlay Networking separation is often done through encapsulation of Tenants traffic using overlay protocols e.g. through VxLAN or EVPN on the Underlay Networks e.g. based on L2 (VLAN) or L3 (IP) networks.

Reviewer: two "e.g." makes it confusing
```suggestion
The Overlay Networking separation is often done through encapsulation of Tenant traffic using overlay protocols such as VxLAN or EVPN over the Underlay Networks (e.g. L2 (VLAN) or L3 (IP) networks).
```

> In some instances, the Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively take over the Physical and Virtual Functions of the NIC directly into the VNF/CNF Tenant. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.

Reviewer: ```suggestion
In some instances, the Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively directly connects the Physical and Virtual Functions of the NIC to the Tenant VNF/CNF. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.
```

> > **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and available standards. Until suitable technologies and standards are developed, CNTT have a set of agreed exemptions that forces the Underlay Networking to handle the bypassed Overlay Networking separation.

Reviewer: ```suggestion
> **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and available standards. Until suitable technologies and standards are developed, CNTT has specified a set of exemptions that allows the Underlay Networking to handle the bypassed Overlay Networking separation.
```

> VTEP could be manually provisioned in the Underlay Networking or be automated and controlled through a Software Defined Networking controller interfaces into the underlying networking in the HW Infrastructure Layer. 

Reviewer: ```suggestion
VTEP could be manually provisioned in the Underlay Networking or be automatically provisioned and managed through Software Defined Networking (SDN) controller interfaces into the underlying networking in the HW Infrastructure Layer. 
```
